src/commands/logs.ts

```diff
@@ -43,12 +47,90 @@ export default class Logs extends Command {
     parse: (value: string): string => value.toLowerCase(),
   }];
 
-  async run(): Promise<void> {
-    inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'));
+  private createLogger(display_name: string) {
+    const { args, flags } = this.parse(Logs);
+    const displayRawLogs = flags.raw || !process.stdout.isTTY;
+    let show_header = true;
+    const prefix = flags.raw ? '' : `${chalk.cyan(chalk.bold(display_name))} ${chalk.hex('#D3D3D3')('|')}`;
+    const columns = process.stdout.columns - (display_name.length + 3);
+
+    function chunkSubstring(str: string, size: number) {
+      const numChunks = Math.ceil(str.length / size);
+      const chunks = new Array(numChunks);
+
+      for (let i = 0, o = 0; i < numChunks; ++i, o += size) {
+        chunks[i] = str.substring(o, o + size);
+      }
+
+      return chunks;
+    }
+
+    return (txt: string) => {
+      if (displayRawLogs) {
+        this.log(txt);
+      } else {
+        if (show_header) {
+          this.log(chalk.bold(chalk.white('Logs:')));
+          this.log(chalk.bold(chalk.white('―'.repeat(process.stdout.columns))));
+          show_header = false;
+        }
+        for (const chunk of chunkSubstring(txt, columns)) {
+          this.log(prefix, chalk.cyan(chunk));
+        }
+      }
+    };
+  }
+
+  async runLocal(): Promise<void> {
+    const { args, flags } = this.parse(Logs);
+
+    const environment_name = await DockerComposeUtils.getLocalEnvironment(this.app.config.getConfigDir(), flags.environment);
+    const compose_file = DockerComposeUtils.buildComposeFilepath(this.app.config.getConfigDir(), environment_name);
+    const service_name = await DockerComposeUtils.getLocalServiceForEnvironment(environment_name, compose_file, args.resource);
+
+    const compose_args = ['-f', compose_file, '-p', environment_name, 'logs'];
```

Comment:
Nice job! Tested it out a bit locally and found some quirks:
1. I can get logs for services/environments that aren't running
    - Don't show environments that aren't running
    - Don't show services that aren't running
2. Not all logs showing when running with and without --follow
   - ex. architect logs -e architect cloud-api
   - I'm not seeing some logs towards the end or when I hot reload our app

Reply:
Actually I'm not so sure about my stance on the not showing services that aren't running. Since a service might start and die and you'd want to see the logs. Maybe make it more clear that the service died or isn't running? Thoughts?

Reply:
I can pull the status field out and display that as well so it would look like
```
cloud-api (RUNNING)
```
Which would make it more clear. Though I do like the idea of showing all services. If they are part of the local environment then I most likely want to interact with them in some way shape or form and even just knowing they are dead is useful.

Will look into the other issues as I did not hit those.

Reply:
@tjhiggins few things
1) An environment is considered to be live if the docker compose yml file exists in the .config folder. I would prefer to cleanup that file on stop rather than do some weird checks to see whats live. Also would prefer to use the .config folder as a source of truth since other things may be running in local docker instance that we don't control.
2. I cannot replicate the log issue, can you share an example so I can get some insight?

Reply:
1. Sounds good to me

2. For the logs issue I'm running:
   - `architect deploy -l auth0.architect.yml -e architect -v values.yml`
   - `architect logs --follow -e architect cloud-api`

Reply:
Found the log issue. For some reason execa is eating parts of stdout. Have no idea why, but if we use spawn everything works perfectly fine. Current version fixes the issue.